bob-backend/src/pearl/data.rs

```diff
@@ -27,6 +35,25 @@ impl AsRef<Key> for Key {
     }
 }
 
+impl PartialOrd for Key {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        use std::cmp::Ordering;
+        for (l, r) in self.0.iter().zip(other.0.iter()).rev() {
```

Comment:
Comparison is a very hot method, so it should be as efficient as possible. Simple for-loop should obviously be much faster

Reply:
Both methods (iteration over the indices or iteration over zipped vecs) produces the same results, according to benchmarks, since Rust compiler has advanced iterator chains processing, and we obviously don't need bound checks here. But I've changed it to indexed search just in case